### PR TITLE
Bootstrapper script for submitting a JAR and running a DAG from it

### PR DESCRIPTION
**Pull request by cangencer**
Additionally, start and stop scripts have been refactored and renamed

### CODE REVIEW

File: hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetBootstrap.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import com.hazelcast.jet.stream.IStreamMap;
+
+import java.lang.reflect.Method;
+import java.util.function.Supplier;
+import java.util.jar.JarFile;
+
+/**
+ * A Bootstrapper for {@link JetInstance} which returns a bootstrapped Jet client instance.
+ * When used in combination with {@code jet-submit.sh}, it runs the JAR file
+ * supplied to {@code jet-submit.sh}. The wrapped instance will then add
```
Comment (viliam-durina) [inline hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetBootstrap.java]: I'd replace "runs" with "runs the Main-Class defined in the manifest in the JAR file"

File: hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetBootstrap.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+ * the given JAR file to all jobs submitted through that instance.
+ * <p>
+ * In the following example, the JAR file containing {@code CustomJetJob}
+ * will be submitted automatically in the {@link JetInstance#newJob(DAG)} method call as part of the
+ * {@link JobConfig} when run as {@code jet-submit.sh custom-jet-job.jar}.
+
```
Comment (viliam-durina) [inline hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetBootstrap.java]: I'd also explain here, WHY we need such utility, that it's the `addClass` method, that needs to be called.

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/ConcurrentMemoizingSupplier.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+             synchronized (this) {
+                 if (remembered == null) {
+                     remembered = onceSupplier.get();
+                     if (remembered == null) {
+                         throw new NullPointerException("Supplier returned null.");
+                     }
+                 }
+             }
+         }
+        return remembered;
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/ConcurrentMemoizingSupplier.java]: Looks like bad indentation here.

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/ConcurrentMemoizingSupplier.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.util;
+
+import java.util.function.Supplier;
+
+/**
+ * Wraps a {@code Supplier} and returns a thread-safe memoizing supplier which calls
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/ConcurrentMemoizingSupplier.java]: Can you limit the text line length to 72 chars here? (I mean the net length of the text line without `* `).

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/Util.java [hunk 5]
```diff
@@ -52,6 +52,10 @@ private Util() {
         return new MemoizingSupplier<>(onceSupplier);
     }
 
+    public static <T> Supplier<T> concurrentMemoize(Supplier<T> onceSupplier) {
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/Util.java]: `memoizeConcurrent` sounds better to me, putting the key word in front.

File: hazelcast-jet-distribution/src/main/resources/bin/jet-submit.sh [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
```
Comment (mtopolnik) [inline hazelcast-jet-distribution/src/main/resources/bin/jet-submit.sh]: Shouldn't we name this `submit-jet.sh` so it's consistent with the other names?

**Comment by mtopolnik** [comment]
verify

**Comment by mtopolnik** [comment]
verify

**Pull request merged**